libs/ledgerjs/packages/context-module/src/external-plugin/data/HttpExternalPluginDataSource.ts

```diff
@@ -0,0 +1,42 @@
+import axios from "axios";
+import { ExternalPluginDataSource, GetDappInfos } from "./ExternalPluginDataSource";
+import { DappDTO } from "./DappDTO";
+import { DappInfos } from "../model/DappInfos";
+import { SelectorDetails } from "../model/SelectorDetails";
+
+export class HttpExternalPluginDataSource implements ExternalPluginDataSource {
+  constructor() {}
+
+  async getDappInfos({ chainId, address, selector }: GetDappInfos): Promise<DappInfos | undefined> {
+    const dappInfos = await axios.request<DappDTO[]>({
+      method: "GET",
+      url: "https://crypto-assets-service.api.ledger.com/v1/dapps",
+      params: { output: "b2c,b2c_signatures,abis", chain_id: chainId, contracts: address },
+    });
+
+    const { erc20OfInterest, method, plugin } =
+      dappInfos.data[0]?.b2c.contracts?.[0]?.selectors?.[selector] || {};
+    const { signature, serialized_data: serializedData } =
+      dappInfos.data[0]?.b2c_signatures?.[address]?.[selector] || {};
+
+    if (!erc20OfInterest || !method || !plugin || !signature || !serializedData) {
+      return;
+    }
+
+    const abi = dappInfos.data[0].abis?.[address];
+
+    if (!abi) {
+      return;
+    }
```

Comment:
Not important for step 1, but I just hear from Alexis that "some contracts could not have an ABI in CAL", which would technically be possible I guess if we don't need to be looking for ERC20 addresses in the calldata 🤔 

Just to keep in mind while we test the module if there is cases of plugin where only `serializedData` & `signature` would be provided 🤷‍♂️